bldsys/cmake/sample_helper.cmake

```diff
@@ -70,6 +73,14 @@ function(add_sample_with_tags)
         endforeach()        
     endif()
 
+    # Add HLSL shader files for this sample
+    if (TARGET_SHADER_FILES_HLSL)
+        list(APPEND SHADER_FILES_HLSL ${TARGET_SHADER_FILES_HLSL})
```

Comment:
Is this intermediate list `SHADER_FILES_HLSL` needed at all? You should be able to directly iterate over `${TARGET_SHADER_FILES_HSLS}`
Note: the same holds for the `SHADER_FILES_GLSL` above.

Reply:
Note that this step is also converting from relative paths to absolute paths, which has to happen at some stage in the process.

Reply:
But that conversion is done in the loop right after this line.
So, essentially, instead of
```
    if (TARGET_SHADER_FILES_GLSL)    
        list(APPEND SHADER_FILES_GLSL ${TARGET_SHADER_FILES_GLSL})
        foreach(SHADER_FILE_GLSL ${SHADER_FILES_GLSL})
            list(APPEND SHADERS_GLSL "${PROJECT_SOURCE_DIR}/shaders/${SHADER_FILE_GLSL}")
        endforeach()        
    endif()
```
One could just write (slightly more efficient as well):
```
    foreach(SHADER_FILE_GLSL ${TARGET_SHADER_FILES_GLSL})
        list(APPEND SHADERS_GLSL "${PROJECT_SOURCE_DIR}/shaders/${SHADER_FILE_GLSL}")
    endforeach()        
```

Reply:
Thanks for code snippet @asuessenbach, I did as you suggested and it seems to work the same.